Enumerables

MarioDena opened the pull request:
Finished Main methods for the project.

It includes a test suite freely customizable.

MarioDena commented on enumerables.rb:
```diff
@@ -0,0 +1,137 @@
+      yield(ar[i])
+      i += 1
+    end
+  end
```
Fixed

MarioDena commented on enumerables.rb:
```diff
@@ -0,0 +1,137 @@
+      i += 1
+    end
+    true
+  end
```
Fixed

MarioDena commented on enumerables.rb:
```diff
@@ -0,0 +1,137 @@
+      i += 1
+    end
+    false
+  end
```
Fixed

MarioDena commented on enumerables.rb:
```diff
@@ -0,0 +1,137 @@
+      i += 1
+    end
+    true
+  end
```
Fixed

MarioDena commented on enumerables.rb:
```diff
@@ -0,0 +1,137 @@
+    true
+  end
+
+  def my_count(lit = '')
```
Fixed

MarioDena commented on enumerables.rb:
```diff
@@ -0,0 +1,137 @@
+    end
+    injector_result
+  end
+end
```
Fixed

MarioDena commented on enumerables.rb:
```diff
@@ -0,0 +1,137 @@
+    ar = self
+    i = 0
+    count = 0
+    if block_given? == false
```
I have no idea most of my code gets formatted by rubocop locally at first =)

MarioDena commented:
Great review @bolah2009  
I've fixed all the issues and submitted a new code review.

Cheers!

stratospherique commented on enumerables.rb:
```diff
@@ -0,0 +1,207 @@
+p multiply_els([1, 2, 3])
+
+# proc map method
+p [1, 2, 3, 4, 5].my_map(&my_proc)
```
- **[REQUIRED]** No debugging instructions in commited code (as discussed [here](https://questions.microverse.org/t/debugging-instructions-code/346))

MarioDena replied:
Those are not debugging instructions, but a test suite. The difference can be found here: https://www.geeksforgeeks.org/differences-between-testing-and-debugging/

stratospherique commented on enumerables.rb:
```diff
@@ -0,0 +1,207 @@
+      i += 1
+    end
+    true
+  end
```
* [ ] enumerables my_all? when a class is passed as an argument returns true if all of the collection is a member of such class
  ```
    array = [1,2,5,4,7]
    p array.my_all?(Integer) == array.all?(Integer) # false
  ```
* [ ] enumerables my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
  ```
    array = %w[dog door rod blade]
    p array.my_all?(/d/) == array.all?(/d/) # false
  ```
* [ ] enumerables my_all? when no block or argument is given returns true when none of the collection members are false or nil
  ```
    true_array = [1, true, 'hi', []]
    p true_array.my_all? == true_array.all? # false
  ```
* [ ] enumerables my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
  ```
    array = [3,3,3,3,3]
    p array.my_all?(3) == array.all?(3) # false
  ```

MarioDena replied:
Fixed

stratospherique commented on enumerables.rb:
```diff
@@ -0,0 +1,207 @@
+  end
+
+  def my_any?(*args)
+    i = 0
```
- Same Issues as my_all?.

MarioDena replied:
fixed

stratospherique commented on enumerables.rb:
```diff
@@ -0,0 +1,207 @@
+
+  def my_none?(*args)
+    i = 0
+    ar = self
```
- Same Issues as my_all?.

MarioDena replied:
fixed

stratospherique commented on enumerables.rb:
```diff
@@ -0,0 +1,207 @@
+
+    injector_result
+  end
+end
```
* [ ] enumerables my_inject combines all elements of enum by applying a binary operation, specified by a block
  ```
    range = Range.new(5,50)
    p range.my_inject(4) { |prod, n| prod * n } 
    # return this errors 
    NoMethodError:
        undefined method `[]' for 5..50:Range
      # ./enumerables.rb:165:in `my_inject'
  ```


* [ ] enumerables my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method
  ```
    range = Range.new(5,50)
    p range.my_inject(2, :*) 
    # return this errors 
    NoMethodError:
        undefined method `[]' for 5..50:Range
      # ./enumerables.rb:163:in `my_inject'

  ```

MarioDena replied:
fixed

MarioDena commented:
@stratospherique  
Fixed all instances, except for 
"[REQUIRED] No debugging instructions in committed code (as discussed here)" As I didn't include debugging instructions please refer to my comment.

Changed the target branch for merge to "dev" as to follow git flow

New code request submitted.

Cheers!
